### PR TITLE
Disable subobjects by default on new packages and data streams

### PR DESCRIPTION
**Pull request by jsoriano**
Modify the archetypes used by `elastic-package create` to disable subobjects by default on new packages and data streams.

A new question is added to the survey so users can chose to opt out for this behavior.

Closes https://github.com/elastic/elastic-package/issues/1872.

### CODE REVIEW

File: internal/packages/archetype/_static/package-manifest.yml.tmpl [hunk 1]
```diff
@@ -72,6 +72,22 @@ policy_templates:
...
 {{ end }}
 {{ end -}}
+{{ if .Manifest.Elasticsearch }}
+elasticsearch:
+{{ if .Manifest.Elasticsearch.SourceMode }}
+  source_mode: {{ .Manifest.Elasticsearch.SourceMode }}
+{{- end}}
+{{ if .Manifest.Elasticsearch.IndexMode }}
+  index_mode: {{ .Manifest.Elasticsearch.IndexMode }}
+{{- end }}
```
Comment (mrodm) [inline internal/packages/archetype/_static/package-manifest.yml.tmpl:82]: Currently, there are no questions for these values when creating the package. So, right now they would not be written here in this manifest.

Should they be added also as part of the questions?

Comment (jsoriano) [reply internal/packages/archetype/_static/package-manifest.yml.tmpl:82]: Not sure, I could add them in a follow up if necessary. Or would you prefer the questions to be added here?

Comment (mrodm) [reply internal/packages/archetype/_static/package-manifest.yml.tmpl:82]: Sure, they can be added in a follow up, it's not needed to do it now.

**Review by mrodm — APPROVED** [review]
:+1:

**Pull request merged**